app/assets/javascripts/govuk_publishing_components/components/show-password.js

```diff
@@ -65,5 +66,26 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     this.input.setAttribute('type', 'password')
   }
 
+  ShowPassword.prototype.moveDataAttributesToButton = function () {
+    var attrs = []
```

Comment:
I would suggest that we can achieve the intended result using only one `for` loop here, by doing something like:

    
    var inputAttrs = this.input.attributes
    for (var j = inputAttrs.length; j >= 0; j--) {
      var attr = inputAttrs[j]
      if (attr && /^data-button/.test(attr.name)) {
        this.showHide.setAttribute(inputAttrs[j].name.replace('-button', ''), inputAttrs[j].value)
        this.input.removeAttribute(inputAttrs[j].name)
      }
    }

Reply:
Ah, of course - going backwards through them means the current index doesn't change when one is deleted. Brilliant, thanks.